remote-api/src/test/java/org/alfresco/rest/api/impl/CategoriesImplTest.java

```diff
@@ -366,19 +379,165 @@ public void testCreateCategories_nonExistingParentNode()
+        assertThat(actualCategory)
+            .isNotNull().usingRecursiveComparison()
+            .isEqualTo(expectedCategory);
+    }
+
+    @Test
+    public void testUpdateCategoryById_noPermission()
+    {
+        given(authorityServiceMock.hasAdminAuthority()).willReturn(false);
+
+        // when
+        assertThatExceptionOfType(PermissionDeniedException.class).isThrownBy(() -> objectUnderTest.updateCategoryById(CATEGORY_ID, CATEGORY));
+
+        then(nodesMock).shouldHaveNoInteractions();
+        then(nodeServiceMock).shouldHaveNoInteractions();
+    }
+
+    @Test
+    public void testUpdateCategoryById_categoryNodeNotFound()
+    {
+        given(nodesMock.validateOrLookupNode(any(), isNull())).willThrow(EntityNotFoundException.class);
+
+        // when
+        assertThatExceptionOfType(EntityNotFoundException.class).isThrownBy(() -> objectUnderTest.updateCategoryById(CATEGORY_ID, CATEGORY));
+
+        then(nodeServiceMock).shouldHaveNoInteractions();
+    }
+
+    @Test
+    public void testUpdateCategoryById_notACategory()
+    {
+        given(nodesMock.isSubClass(any(), any(), eq(false))).willReturn(false);
+
+        // when
+        assertThatExceptionOfType(InvalidArgumentException.class).isThrownBy(() -> objectUnderTest.updateCategoryById(CATEGORY_ID, CATEGORY))
+            .withMessageContaining(NOT_A_VALID_CATEGORY);
+
+        then(nodeServiceMock).shouldHaveNoInteractions();
+    }
+
+    @Test
+    public void testUpdateCategoryById_isRootCategory()
+    {
+        given(nodesMock.validateOrLookupNode(any(), isNull())).willReturn(createNodeRefUsingId(CAT_ROOT_NODE_ID));
+        given(nodeServiceMock.getParentAssocs(any())).willReturn(List.of(categoryChildAssociationRefMock));
+        given(categoryChildAssociationRefMock.getQName()).willReturn(ContentModel.ASPECT_GEN_CLASSIFIABLE);
+
+        // when
+        assertThatExceptionOfType(InvalidArgumentException.class).isThrownBy(() -> objectUnderTest.updateCategoryById(CATEGORY_ID, CATEGORY))
+            .withMessageContaining(NOT_A_VALID_CATEGORY);
+    }
+
+    @Test
+    public void testUpdateCategoryById_emptyName()
+    {
+        final Category categoryWithoutName = createCategoryWithName("");
+        final QName categoryQName = createCmQNameOf(CATEGORY_NAME);
+        final NodeRef parentCategoryNodeRef = createNodeRefUsingId(PARENT_ID);
+        final ChildAssociationRef parentAssociation = createAssociationOf(parentCategoryNodeRef, CATEGORY_NODE_REF, categoryQName);
+        given(nodesMock.getNode(any())).willReturn(prepareCategoryNode());
+        given(nodeServiceMock.getPrimaryParent(any())).willReturn(parentAssociation);
+
+        // when
+        assertThatExceptionOfType(InvalidArgumentException.class).isThrownBy(() -> objectUnderTest.updateCategoryById(CATEGORY_ID, categoryWithoutName))
+            .withMessageContaining(NOT_NULL_NOR_EMPTY);
+    }
+
+    @Test
+    public void testUpdateCategoryById_notMatchingIdField()
```

Comment:
As I commented in the tested class - I believe we should either ignore all fields other than name or shouldn't allow any additional fields.

Reply:
I think we should ignore them - at least I'm pretty sure that's what happens in other PUT APIs.